internal/api/api.go

```diff
@@ -90,7 +90,7 @@ func HTTPServer(
 	handler = versionHeaderMiddleware(config.AppVersion)(handler)
 	if config.Debug {
 		handler = goahttpmwr.Log(loggerAdapter(logger))(handler)
-		handler = goahttpmwr.Debug(mux, os.Stdout)(handler)
+		handler = Debug(mux, os.Stdout)(handler)
```

Comment:
Is it possible to wrap `goahttpmwr.Debug` with an additional layer that checks the content type and conditionally intercepts the request? If the request content type indicates a large body,  you could temporarily replace the body with an empty reader, allowing it to pass through the middleware without printing the full body. After the middleware processes the request, then reassign the original body for further processing. Not sure if that's possible but it seems doable.

Reply:
```go
func conditionalDebug(next http.Handler) http.Handler {
    return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
        if r.Header.Get("Content-Type") == "..." {
            originalBody := r.Body
            r.Body = io.NopCloser(io.LimitReader(originalBody, 0))
            next.ServeHTTP(w, r)
            r.Body = originalBody
        } else {
            next.ServeHTTP(w, r)
        }
    })
}

handler = conditionalDebug(goahttpmwr.Debug(mux, os.Stdout)(handler))
```

Reply:
It's really not that simple because this middleware: 
 - Intercepts the (outgoing request), writes info to the buffer
 - calls next.ServeHTTP (in the middle of the function
 - When the `ServerHTTP` returns then it works on the response.

Reply:
This solution is better but it still does not fit the optimal solution for the client. Fundamentally the client wants be able to Debug the API when invoking the Download package endpoint, and by Debug we mean to be able to show request/response headers, params, status, etc.. without printing the response body (because is a binary blob, we don't care about that). 

Something like this would be the expected output in stdout: 

```sh
> [VVbQVl5C] GET /collection/3/download
> [VVbQVl5C] Accept: */*
> [VVbQVl5C] User-Agent: curl/8.10.1
< [VVbQVl5C] OK
< [VVbQVl5C] Content-Disposition: attachment; filename="transfer_2_1.zip-830237d2-d82b-4943-b561-7fccac2ada1f.7z"
< [VVbQVl5C] Content-Length: 200895757
< [VVbQVl5C] Content-Type: application/x-7z-compressed
< [VVbQVl5C] X-Enduro-Version: (dev-version)
```
Note that we print all information, skipping only the body being printed (but successfully downloaded).

Your current proposal will either print all the Debug information (including the body) or nothing at all. Basically the `io.Writer` that the middleware takes will write everything. That is why I decided to fork/vendor this specific middleware, I could not see how to achieve the objective without that approach.